front/cypress/e2e/admin/internal_comments/initiative_internal_comments.cy.ts

```diff
@@ -0,0 +1,110 @@
+      initiativeId2 = initiaitve.body.data.id;
+    });
+  });
+
+  it('allows users to post, edit and delete an internal comment on an initiative', () => {
+    const internalComment = randomString();
+    const editedComment = randomString();
+
+    cy.setAdminLoginCookie();
+    cy.visit(`admin/initiatives/${initiativeId1}`);
+    cy.acceptCookies();
+
+    // Create comment and check that comment is created
+    cy.get('[data-cy="e2e-tab-internal"]').click();
+    cy.get('#submit-comment').should('exist');
+    cy.get('#submit-comment').click().type(internalComment);
+    cy.get('.e2e-submit-parentcomment').click();
+    cy.get('.e2e-parentcomment').contains(internalComment);
+
+    // Edit comment and check that comment is edited
+    cy.get('[data-testid="moreOptionsButton"]').first().should('exist');
+    cy.get('[data-testid="moreOptionsButton"]').first().click();
+    cy.get('.e2e-more-actions-list button').eq(1).contains('Edit');
+    cy.get('.e2e-more-actions-list button').eq(1).click();
+    cy.get('#e2e-internal-comment-edit-textarea')
+      .click()
+      .clear()
+      .type(editedComment);
+    cy.get('#e2e-save-internal-comment-edit-button').click();
+    cy.get('.e2e-parentcomment').contains(editedComment);
+
+    // Delete comment and check that comment is deleted
+    cy.get('[data-testid="moreOptionsButton"]').first().should('exist');
+    cy.get('[data-testid="moreOptionsButton"]').first().click();
+    cy.get('.e2e-more-actions-list button').eq(0).contains('Delete');
+    cy.get('.e2e-more-actions-list button').eq(0).click();
+    cy.get('#e2e-confirm-deletion').should('exist');
+    cy.get('#e2e-confirm-deletion').click();
+    cy.get('.e2e-parentcomment').should('not.exist');
+  });
+
+  it('allows users to reply to internal comments, edit and delete a reply', () => {
+    const internalComment = randomString();
+    const replyComment = randomString();
+    const editedReply = randomString();
+    cy.setAdminLoginCookie();
+    cy.visit(`admin/initiatives/${initiativeId2}`);
+    cy.acceptCookies();
+    // Create comment and check that comment is created
+    cy.get('[data-cy="e2e-tab-internal"]').click();
+    cy.get('#submit-comment').should('exist');
+    cy.get('#submit-comment').click().type(internalComment);
+    cy.get('.e2e-submit-parentcomment').click();
+    cy.get('.e2e-parentcomment').contains(internalComment);
+
+    // Reply to comment and check that reply is created
+    cy.get('.e2e-comment-reply-button').should('exist');
+    cy.get('.e2e-comment-reply-button').click();
+    cy.get('#e2e-child-comment-text-area').should('exist');
+    cy.get('#e2e-child-comment-text-area').click().type(replyComment);
+    cy.get('.e2e-submit-childcomment').first().click();
+    cy.get('.e2e-childcomment').contains(replyComment);
+
+    // Edit reply and check that reply is edited
+    cy.get('[data-testid="moreOptionsButton"]').eq(1).should('exist');
+    cy.get('[data-testid="moreOptionsButton"]').eq(1).click();
```

Comment:
Not very deep into this so sorry if it's a stupid question but quickly want to check: how do we know this is the more options button of the child comment?

Reply:
It isn't stupid actually. It is about how the test is setup. The second comment will always be the child comment. We should be safe with it but let me make it more robust and easier to follow. Good question